packages/react-components/src/layout/src/Flex.tsx

```diff
@@ -1,102 +1,69 @@
 
 import { Box } from "../../box";
-import { any, bool, elementType, oneOf, oneOfType, string } from "prop-types";
-import { cssModule, mergeProps } from "../../shared";
-import { forwardRef } from "react";
+import { ElementType, ForwardedRef, ReactElement, ReactNode, forwardRef } from "react";
+import { PropsWithoutForwardedRef, cssModule, mergeProps } from "../../shared";
 import { isNil, isString } from "lodash";
 import { useMemo } from "react";
 
-const propTypes = {
+interface FlexProps {
     /**
      * How the elements are placed in the container. See [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/flex-direction).
      */
-    direction: oneOf(["row", "column"]),
+    direction?: "row" | "column";
     /**
      * Whether or not to inline the elements.
      */
-    inline: bool,
+    inline?: boolean;
     /**
      * Whether or not to reverse the order of the elements.
      */
-    reverse: bool,
+    reverse?: boolean;
     /**
      * The distribution of space around child items along the cross axis. See [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/align-content).
      */
-    alignContent: oneOf([
-        "start",
-        "end",
-        "center",
-        "space-between",
-        "space-around",
-        "space-evenly",
-        "stretch",
-        "baseline",
-        "first baseline",
-        "last baseline",
-        "safe center",
-        "unsafe center"
-    ]),
+    alignContent?: "start" | "end" | "center" | "space-between" | "space-around" | "space-evenly" | "stretch" | "baseline" | "first baseline" | "last baseline" | "safe center" | "unsafe center";
     /**
      * The alignment of children within their container. See [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/align-items).
      */
-    alignItems: oneOf([
-        "start",
-        "end",
-        "center",
-        "stretch",
-        "self-start",
-        "self-end",
-        "baseline",
-        "first baseline",
-        "last baseline",
-        "safe center",
-        "unsafe center"
-    ]),
+    alignItems?: "start" | "end" | "center" | "stretch" | "self-start" | "self-end" | "baseline" | "first baseline" | "last baseline" | "safe center" | "unsafe center";
     /**
      * The distribution of space around items along the main axis. See [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/justify-content).
      */
-    justifyContent: oneOf([
-        "start",
-        "end",
-        "center",
-        "left",
-        "right",
-        "space-between",
-        "space-around",
-        "space-evenly",
-        "stretch",
-        "baseline",
-        "first baseline",
-        "last baseline",
-        "safe center",
-        "unsafe center"
-    ]),
+    justifyContent?: "start" | "end" | "center" | "left" | "right" | "space-between" | "space-around" | "space-evenly" | "stretch" | "baseline" | "first baseline" | "last baseline" | "safe center" | "unsafe center"
+    /**
+     * TODO: document this property
```

Comment:
This props was not documented. Do we leave TODO here and come back at the end of the conversion?

Reply:
sure leave it there.